src/artemisa/reducer.js

```diff
@@ -11,4 +12,46 @@ export function artemisa(state = {}, action) {
     return processApiAction(storagePropertyNameForAction(action), state, action)
   }
   return state
-}
\ No newline at end of file
+}
+
+export function processApiAction(stateProperty, state, action) {
+  if (isRequest(action)) {
+    return {
+      ...state,
+      [stateProperty]: {
+        state: State.FETCHING,
+        path: action.path
+      }
+    }
+  }
+  if (isReceive(action)) {
+    const { [stateProperty]: myState, ...restOfState } = state
+    return (!myState || myState.path === action.path)
+      ? {
+        ...restOfState,
+        [stateProperty]: {
+          state: State.FETCHED,
+          path: action.path,
+          value: action.data
+        }
+      }
+      // Received action does not correspond to the path in FETCHING state,
+      // we asume that the data received corresponds to an old request is old and discard it.
+      : state
+  }
+  if (isError(action)) {
+    return {
+      ...state,
+      [stateProperty]: {
+        state: State.ERROR,
+        path: action.path,
+        error: action.error
+      }
+    }
+  }
+  throw new Error('Unknown API action', action)
+}
```

Comment:
Also took this from `core/reducer` since the actual reducer that user will use is this one (artemisa) and not the old one that was for decorating user's reducers. (sounds like a riddle)